src/appleseed.studio/utility/miscellaneous.cpp

```diff
@@ -183,6 +183,18 @@ namespace
     }
 }
 
+QString get_extension(ParamArray& settings, const QString& target_dialog) {
+    QString filter = get_value(settings, target_dialog + SETTINGS_SELECTED_FILTER);
+
+    const int ext_start = filter.lastIndexOf('(') + 2;
```

Comment:
No, it's not appending anything. You can see it in mainwindow.cpp file, where get_extension used.
On 1237 line filepath is retrieved from file dialog, and it's just like it was printed in dialog. 
It can be like "a.appleseed" and if on line 1247 will return false and no extension will be added. Otherwise in case of just "a" extension chosen in dialog is retrieved and appended.